modules/Microsoft.Synapse/workspaces/deploy.bicep

```diff
@@ -15,6 +15,9 @@ param azureADOnlyAuthentication bool = false
 @description('Optional. AAD object ID of initial workspace admin.')
 param initialWorkspaceAdminObjectID string = ''
 
+@description('Required. Resource ID of the default ADLS Gen2 storage account.')
+param defaultDataLakeStorageAccountResourceId string
```

Comment:
Is there any constraint to where the storage should be placed? Is it required to be in the same subscription and resource group as the main resource?
If so, we could derive the resource id from the input name as an existing resource. Otherwise, 
I guess we should get rid of the below storage account name and derive it from the new resource ID input parameter. What's your take on that?

Reply:
Thanks for the suggestion @eriqua . I removed the `defaultDataLakeStorageAccountName` and calculating the name using `defaultDataLakeStorageAccountResourceId`.